src/core/utils/hooks/useResizeObserver.tsx

```diff
@@ -0,0 +1,50 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+import React from 'react';
+
+/**
+ * Hook that uses `ResizeObserver` to return an element's size every time it updates.
+ * @param elementRef ref of the element to observe resizes on.
+ * @param delay maximum timeout in ms to debounce the update by.
+ * @returns stateful object containing height and width of the element.
+ */
+export const useResizeObserver = <T extends HTMLElement>(
+  elementRef: React.RefObject<T>,
+  { delay = 0 } = {},
+) => {
+  const [size, setSize] = React.useState({
```

Comment:
This init code will be called always, so you can change it to function. That way it will be initialized only once.

Reply:
Changed